Branch-SDK/src/main/java/io/branch/referral/AttributionReportingManager.kt

```diff
@@ -0,0 +1,166 @@
+                    val params = getParams(context) + "&conversion_id=$conversionId"
+                    val branchBaseURL = PrefHelper.getInstance(context).apiBaseUrl
+                    val triggerUri = Uri.parse("${branchBaseURL}${Defines.RequestPath.RegisterTrigger}?$params")
+                    val manager = MeasurementManager.get(context)
+
+                    manager.registerTrigger(triggerUri, executor, object : OutcomeReceiver<Any?, Exception> {
+                            override fun onResult(result: Any?) {
+                                BranchLogger.v("Trigger registered successfully with URI: $triggerUri")
+                                executor.shutdown()
+                            }
+
+                            override fun onError(e: Exception) {
+                                BranchLogger.w("Error while registering trigger: ${e.message}")
+                                executor.shutdown()
+                            }
+                        }
+                    )
+                } else {
+                    BranchLogger.v("Measurement API is not enabled. Did not register trigger.")
+                }
+            }
+        }
+    }
+
+    fun getParams(context: Context): String {
+        val systemObserver = DeviceInfo.getInstance().systemObserver
+        val prefHelper = PrefHelper.getInstance(context)
+
+        val gaid = systemObserver.aid
+        val hardwareId = SystemObserver.getUniqueID(context, false).id
+        val brand = SystemObserver.getPhoneBrand()
+        val model = SystemObserver.getPhoneModel()
+        val osVersion = SystemObserver.getOSVersion()
+        val apiLevel = SystemObserver.getAPILevel()
+        val localIp = SystemObserver.getLocalIPAddress()
+        val connectionType = SystemObserver.getConnectionType(context)
+        val carrier = SystemObserver.getCarrier(context)
+        val screenDisplay = SystemObserver.getScreenDisplay(context)
+        val dpi = screenDisplay.densityDpi
+        val screenWidth = screenDisplay.widthPixels
+        val screenHeight = screenDisplay.heightPixels
+        val uiMode = SystemObserver.getUIMode(context)
+        val osName = SystemObserver.getOS(context)
+        val language = SystemObserver.getISO2LanguageCode()
+        val country = SystemObserver.getISO2CountryCode()
+        val sandboxVersion = when {
+            Build.VERSION.SDK_INT >= Build.VERSION_CODES.R &&
+                    SdkExtensions.getExtensionVersion(Build.VERSION_CODES.R) >= 4 -> {
+                SdkExtensions.getExtensionVersion(SdkExtensions.AD_SERVICES)
+            }
+            else -> 0  // Default to 0 if conditions are not met
+        }
+        val appPackageName = context.packageName
+        val timestamp = System.currentTimeMillis()
+        val platform = "ANDROID_APP"
+        val os = "ANDROID"
+        val appVersion = SystemObserver.getAppVersion(context)
+        val eea = prefHelper.eeaRegion
```

Comment:
What assumptions are being made here? Does this code assume that it will have all signals ready before being called? Mention in comments